lib/runtime-core/src/backing.rs

```diff
@@ -163,141 +165,134 @@ impl LocalBacking {
 
             match init.table_index.local_or_import(module) {
                 LocalOrImport::Local(local_table_index) => {
-                    let table = &mut tables[local_table_index];
-                    match table.elements {
-                        TableElements::Anyfunc(ref mut elements) => {
-                            if elements.len() < init_base + init.elements.len() {
-                                // Grow the table if it's too small.
-                                elements
-                                    .resize(init_base + init.elements.len(), vm::Anyfunc::null());
-                            }
-
-                            for (i, &func_index) in init.elements.iter().enumerate() {
-                                let sig_index = module.func_assoc[func_index];
-                                let sig_id = vm::SigId(sig_index.index() as u32);
-
-                                let func_data = match func_index.local_or_import(module) {
-                                    LocalOrImport::Local(local_func_index) => vm::ImportedFunc {
-                                        func: module
-                                            .func_resolver
-                                            .get(module, local_func_index)
-                                            .unwrap()
-                                            .as_ptr(),
-                                        vmctx,
-                                    },
-                                    LocalOrImport::Import(imported_func_index) => {
-                                        imports.functions[imported_func_index].clone()
-                                    }
-                                };
-
-                                elements[init_base + i] = vm::Anyfunc { func_data, sig_id };
-                            }
-                        }
+                    let table = &tables[local_table_index];
+
+                    if (table.current_size() as usize) < init_base + init.elements.len() {
+                        let delta =
+                            (init_base + init.elements.len()) - table.current_size() as usize;
+                        // Grow the table if it's too small.
+                        table.grow(delta as u32).expect("couldn't grow table");
```

Comment:
Maybe a `RangeError` in the future?

Reply:
Yeah, definitely errors need to better. I left them out in a lot of cases.

Reply:
I've added new errors.